arch/arc/core/fault.c

```diff
@@ -29,6 +31,99 @@ static const struct z_exc_handle exceptions[] = {
 };
 #endif
 
+#if defined(CONFIG_MPU_STACK_GUARD)
+
+#define IS_MPU_GUARD_VIOLATION(guard_start, fault_addr, stack_ptr) \
+	((fault_addr >= guard_start) && \
+	(fault_addr < (guard_start + STACK_GUARD_SIZE)) && \
+	(stack_ptr <= (guard_start + STACK_GUARD_SIZE)))
+
+/**
+ * @brief Assess occurrence of current thread's stack corruption
+ *
+ * This function performs an assessment whether a memory fault (on a
+ * given memory address) is the result of stack memory corruption of
+ * the current thread.
+ *
+ * Thread stack corruption for supervisor threads or user threads in
+ * privilege mode (when User Space is supported) is reported upon an
+ * attempt to access the stack guard area (if MPU Stack Guard feature
+ * is supported). Additionally the current thread stack pointer
+ * must be pointing inside or below the guard area.
+ *
+ * Thread stack corruption for user threads in user mode is reported,
+ * if the current stack pointer is pointing below the start of the current
+ * thread's stack.
+ *
+ * Notes:
+ * - we assume a fully descending stack,
+ * - we assume a stacking error has occurred,
+ * - the function shall be called when handling MemManage and Bus fault,
+ *   and only if a Stacking error has been reported.
+ *
+ * If stack corruption is detected, the function returns the lowest
+ * allowed address where the Stack Pointer can safely point to, to
```

Comment:
This return value doesn't seem to be used on ARC, is the corrupt stack pointer 'repaired' elsewhere? (or simply not used after the stack exception, since exception handling uses a different stack).

Reply:
returning the lowest allowed stack frame pointer sometimes is useful when it's required to do some stack frame analysis,